### ISSUE TITLE
Defining const to nothing

### ISSUE THREAD

**Issue by kaidokert** [madler/zlib#460]
On unrelated work, this piece of code in `zconf.h` tripped really unfun to trace compiler errors:

```
#ifndef STDC
#  ifndef const /* cannot use !defined(STDC) && !defined(const) on Mac */
#    define const       /* note: need a more gentle solution here */
#  endif
#endif
```

That code apparently dates back to Zlib 0.93 (25 June 95), and i would hope is not needed anywhere for anything these days. Regardless of why STDC might end up being undefined at some point during porting, defining language keywords to nothing should probably not be necessary. An `#error "Please have _the talk_ with your compiler"` would maybe suffice ?

**Comment by madler** [comment]
Agreed, we need to do something different here. As noted in the comment.
